### PR TITLE
chore(auth): Update credentials to store signedIn info

### PR DESCRIPTION
**Pull request by royjit**
*Issue #, if available:*

*Description of changes:*

*Check points: (check or cross out if not relevant)*

- [ ] Added new tests to cover change, if needed
- [ ] Build succeeds with all target using Swift Package Manager
- [ ] All unit tests pass
- [ ] All integration tests pass
- [ ] Security oriented best practices and standards are followed (e.g. using input sanitization, principle of least privilege, etc)
- [ ] Documentation update for the change if required
- [ ] PR title conforms to conventional commit style
- [ ] If breaking change, documentation/changelog update with migration instructions

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/StateMachine/CodeGen/Data/HostedUIProviderInfo.swift [hunk 1]
```diff
@@ -28,8 +28,8 @@ extension HostedUIProviderInfo: Codable {
 
     init(from decoder: Decoder) throws {
         let values = try decoder.container(keyedBy: CodingKeys.self)
-        idpIdentifier = try values.decode(String.self, forKey: .idpIdentifier)
-        federationProviderName = try values.decode(String.self, forKey: .federationProviderName)
+        idpIdentifier = try? values.decode(String.self, forKey: .idpIdentifier)
+        federationProviderName = try? values.decode(String.self, forKey: .federationProviderName)
```
Comment (harsh62) [inline AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/StateMachine/CodeGen/Data/HostedUIProviderInfo.swift:32]: what is the reason of this change?

File: AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/StateMachine/CodeGen/Data/SignedInData.swift [hunk 2]
```diff
@@ -15,16 +15,14 @@ struct SignedInData {
...
+    init(signedInDate: Date,
+         signInMethod: SignInMethod,
+         deviceMetadata: DeviceMetadata = .noData,
+         cognitoUserPoolTokens: AWSCognitoUserPoolTokens
     ) {
-        self.userId = userId
-        self.userName = userName
+        let user = try? TokenParserHelper.getAuthUser(accessToken: cognitoUserPoolTokens.accessToken)
+        self.userId = user?.userId ?? "unknown"
+        self.userName = user?.username ?? "unknown"
```
Comment (harsh62) [inline AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/StateMachine/CodeGen/Data/SignedInData.swift:25]: This will eventually be passed back to the customer.. Are we okay passing `unknown` value back?

File: AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/StateMachine/CodeGen/Data/HostedUIProviderInfo.swift (hunk 1 above)
Comment (royjit) [reply AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/StateMachine/CodeGen/Data/HostedUIProviderInfo.swift:32]: these values are optional, but decoding fails if we are expecting non-optional value. So to avoid it changed the try to optional.

Comment (harsh62) [reply AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/StateMachine/CodeGen/Data/HostedUIProviderInfo.swift:32]: We should use `decodeIfPresent` in that case rather than `decode`.

File: AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/StateMachine/CodeGen/Data/SignedInData.swift (hunk 2 above)
Comment (royjit) [reply AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/StateMachine/CodeGen/Data/SignedInData.swift:25]: As discussed this will be always passed back from Cognito. "Unknown" is just a fallback so that we do not need to crash the app, in case parsing fails.

**Review by harsh62 — APPROVED** [review]
